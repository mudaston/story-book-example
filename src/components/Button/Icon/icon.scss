.icon {
	display: flex;
	align-items: center;
	justify-content: center;

	width: 20px;
	height: 20px;

	pointer-events: none;

	img {
		height: auto;
		width: 100%;

		min-height: 100%;
		max-height: 100%;

		min-width: 100%;
		max-width: 100%;

		object-fit: cover;
	}
}
