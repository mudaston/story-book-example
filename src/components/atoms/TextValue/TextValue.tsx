import { FC } from 'react'

import {
	textComponentCanOnlyContainsText,
	textComponentCannotBeEmpty,
} from '../../../errors/TextValue'

interface ValueProps {
	children: String
}

type Props = ValueProps

const TextValue: FC<Props> = ({ children }) => {
	if (!children) throw new Error(textComponentCannotBeEmpty)

	if (typeof children !== 'string')
		throw new Error(textComponentCanOnlyContainsText)

	return <span>{children}</span>
}

TextValue.displayName = 'TextValue'

export default TextValue
