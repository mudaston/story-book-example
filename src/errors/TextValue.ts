const textComponentCannotBeEmpty =
	'Cannot render TextValue component without value'

const textComponentCanOnlyContainsText = 'Cannot render non-string value'

export { textComponentCannotBeEmpty, textComponentCanOnlyContainsText }
