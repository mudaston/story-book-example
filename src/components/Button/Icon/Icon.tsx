import { FC, ReactNode } from 'react'

import './icon.scss'

interface OwnProps {
	children: JSX.Element | ReactNode | String
}

type Props = OwnProps

const Icon: FC<Props> = ({ children }) => {
	return <div className="icon">{children}</div>
}

export default Icon
