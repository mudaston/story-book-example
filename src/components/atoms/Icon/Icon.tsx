import React, { FC } from 'react'

import './Icon.scss'

interface IconProps {
	children: JSX.Element | React.ReactNode
}

type Props = IconProps

const Icon: FC<Props> = ({ children }) => {
	return <span className="izi-icon">{children}</span>
}

export default Icon
