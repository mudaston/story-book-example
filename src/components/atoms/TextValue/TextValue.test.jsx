import { fireEvent, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'

import TextValue from './TextValue'
import Icon from '../Icon'

import {
	textComponentCanOnlyContainsText,
	textComponentCannotBeEmpty,
} from './utils/error'

describe('render TextValue component', () => {
	const text = 'Hello world'

	it('render with value', async () => {
		const component = <TextValue>{text}</TextValue>
		const { asFragment } = render(component)

		expect(screen.getByText(text)).toBeInTheDocument()
		expect(screen.getByText(text)).toHaveTextContent(text)

		expect(asFragment(component)).toMatchSnapshot()
	})

	it('render without value', async () => {
		expect(() => render(<TextValue></TextValue>)).toThrowError(
			textComponentCannotBeEmpty
		)
	})

	it('render with non-string value', async () => {
		expect(() =>
			render(
				<TextValue>
					<Icon />
				</TextValue>
			)
		).toThrowError(textComponentCanOnlyContainsText)
	})
})
