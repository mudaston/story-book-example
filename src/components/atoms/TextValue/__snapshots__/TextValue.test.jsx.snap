// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render TextValue component render with value 1`] = `
<DocumentFragment>
  <span>
    Hello world
  </span>
</DocumentFragment>
`;
